thor/request.py

```diff
@@ -49,39 +75,63 @@ def handleRequest(arguments, ncFileDictTree, log):
 
     # ---------------
     requestedFiles = getList(ncFileDictTree,
-                             domain,
                              model,
                              experiment,
                              variable)
 
+    returnData = np.ndarray(shape=arguments["return-dimension"],
+                            dtype=float)
+    returnData.fill(100000)
+
     for ncFile in requestedFiles:
         if arguments["from-date"] > ncFile.getStartDate()\
-                and arguments["to-date"] < ncFile.getLastDate():
-                    # If climateArea is false, it is not within file
-                    climateAreaDict = ncFile.getData(
-                        arguments["from-date"],
-                        arguments["to-date"],
-                        arguments["from-latitude"],
-                        arguments["to-latitude"],
-                        arguments["from-longitude"],
-                        arguments["to-longitude"])
-
-                    if not climateAreaDict["ok"]:
-                        return climateAreaDict
-
-                    # Interpolating the data
-                    returnAreaDict = sigProcess.interpolate(
-                        climateAreaDict["data"],
-                        arguments["return-dimension"])
-
-                    if not returnAreaDict["ok"]:
-                        return returnAreaDict
-
-                    return {"ok": True,
-                            "data": returnAreaDict["data"]}
-
-    returnDataDict = {"ok": False,
-                      "errorMessage":
-                      "Specified date range not within server dataset."}
-
-    return returnDataDict
+           and arguments["to-date"] < ncFile.getLastDate():
+            climateAreaDict = ncFile.getData(
+                arguments["from-date"],
+                arguments["to-date"],
+                arguments["from-latitude"],
+                arguments["to-latitude"],
+                arguments["from-longitude"],
+                arguments["to-longitude"])
+
+            if climateAreaDict["ok"]:
+                [latScale,
+                 lonScale] = overlapScale(climateAreaDict["area"],
+                                          arguments["from-latitude"],
+                                          arguments["to-latitude"],
+                                          arguments["from-longitude"],
+                                          arguments["to-longitude"])
+
+                latInterpolLen = floor(arguments["return-dimension"][0] *
+                                       latScale)
+                lonInterpolLen = floor(arguments["return-dimension"][1] *
```

Comment:
Flooring here is not necessary since we're checking if return dimension is integer in check-arguments.